src/boot/pe.c

```diff
@@ -441,17 +469,36 @@ EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address, size_t
         if (!verify_pe(dos, pe, /* allow_compatibility= */ true))
                 return EFI_LOAD_ERROR;
 
+        uint64_t image_base;
+        if (pe->OptionalHeader.Magic == OPTHDR32_MAGIC)
+                image_base = pe->OptionalHeader.ImageBase32;
+        else
+                image_base = pe->OptionalHeader.ImageBase64;
+
+        const PeImageDataDirectory *data_directory;
+        if (pe->OptionalHeader.Magic == OPTHDR32_MAGIC)
+                data_directory = pe->OptionalHeader.DataDirectory32;
+        else
+                data_directory = pe->OptionalHeader.DataDirectory64;
+
+        if (data_directory[5].Size != 0)
```

Comment:
where does the 5 come from? please add comment

Reply:
i.e. say that 5 is "Base Relocation Table"

alternatively, define a macro for that

Reply:
hmm, pe_kernel_info() is also used from boot.c (i.e. sd-boot) where we don#t really care for the entrypoint and pass NULL for that. hence, in that case  this check is not necessary, and maybe even detrimental.

hence, please condition this check so that it is only done if caller actually asks for the entrypoint i.e. if ret_address is non-NULL

Reply:
also please add a comment here that explains the check. i.e. that the kernel's own efi stub does not require relocations, and we hence filter this out early. but that we are happy to add relocation code, if this changes at some later point

Reply:
I have split the function instead.

I will add more comments.